Lib/test/test_cmd_line.py

```diff
@@ -971,11 +972,19 @@ def test_python_legacy_windows_fs_encoding(self):
         self.assertIn(expected.encode(), out)
 
     @unittest.skipUnless(support.MS_WINDOWS, 'Test only applicable on Windows')
+    @unittest.skipUnless(type(sys.stdin.buffer.raw) == io._WindowsConsoleIO,
+                         'Requires real console handles')
     def test_python_legacy_windows_stdio(self):
-        code = "import sys; print(sys.stdin.encoding, sys.stdout.encoding)"
-        expected = 'cp'
-        rc, out, err = assert_python_ok('-c', code, PYTHONLEGACYWINDOWSSTDIO='1')
-        self.assertIn(expected.encode(), out)
+        code = "import sys; print(sys.stdin.encoding, sys.stdout.encoding, file=sys.stderr)"
+        env = os.environ.copy()
+        env['PYTHONLEGACYWINDOWSSTDIO'] = '1'
+        # Use Popen directly to ensure stdin/out are console handles
+        p = subprocess.Popen([sys.executable, '-c', code],
+                              stderr=subprocess.PIPE, env=env)
+        out = p.stderr.read()
+        p.stderr.close()
+        p.wait()
+        self.assertNotIn(b'utf-8', out)
```

Comment:
This might not be true. It's entirely possible to set the default encoding to UTF-8 in Windows itself, at which point the legacy stdio will also be UTF-8.

Checking the type of the stdin/stdout objects in the subprocess is probably the best way. The _real_ purpose of this environment variable is to go back to using `FileIO` rather than `_WindowsConsoleIO`, and the encoding is only a side-effect of that.

And if we just make the child process `assert`, then we can check the exit code rather than reading the output.